pydevices/acq400Devices/acq400_base.py

```diff
@@ -284,18 +349,65 @@ class _ACQ400_TR_BASE(_ACQ400_BASE):
     A child class of _ACQ400_BASE that contains the specific methods for
     taking a transient capture.
     """
+    tr_base_parts = [
+        {'path':':RUNNING',     'type':'numeric',                  'options':('no_write_model',)},
+        {'path':':TRIG_TIME',   'type':'numeric',                  'options':('write_shot',)}
+        ]
 
-    def arm(self):
-        import acq400_hapi
+    def trig(self, msg=''):
+        message = str(msg)
+        uut = acq400_hapi.Acq400(self.node.data())
+        self.dprint(1, "The trigger message for transient recording is: %s", message)
+
+        wrtdtx = '1 --tx_id=' + message
+        uut.s0.wrtd_tx_immediate = wrtdtx
+    
+        #self.trig_time.putData(MDSplus.Int64(uut.s0.wr_tai_cur))
+    TRIG=trig
+
+    def state(self):
+        uut = acq400_hapi.Acq400(self.node.data())
+        _status = [int(x) for x in uut.s0.state.split(" ")]
+
+        #statemon = acq400_hapi.acq400.Statusmonitor(self.node.data(), _status)
+        #istate   = int(statemon.get_state())
+
+        trstate = acq400_hapi.acq400.STATE.str(_status[0])
+        
+        print("TR state: {}".format(trstate))
+        print("TR status: {}".format(_status))
+    STATE=state
+    
+    def stop(self):
+        uut = acq400_hapi.Acq400(self.node.data())
+        print("PRE {}, POST {} and ELAPSED {}".format(uut.pre_samples(), uut.post_samples(), uut.elapsed_samples()))
```

Comment:
Change this to use the `%` formatting

Reply:
though out.. after rebasing this should trigger a fail if it is a problem some .format() syntax is ok but this one is not

Reply:
Done